packages/pwa-kit-runtime/src/ssr/server/express.js

```diff
@@ -461,11 +461,23 @@ export const respondFromBundle = ({req, res, path, redirect = 301}) => {
 }
 
 export const getRuntime = () => {
-    return isRemote()
+    const runtime = isRemote()
         ? RemoteServerFactory
         : // The dev server is for development only, and should not be deployed to production.
           // To avoid deploying the dev server (and all of its dependencies) to production, it exists
           // as an optional peer dependency to this package. The unusual `require` statement is needed
           // to bypass webpack and ensure that the dev server does not get bundled.
           eval('require').main.require('pwa-kit-dev/ssr/server/build-dev-server').DevServerFactory
+
+    // The runtime is a JavaScript object.
+    // Sometimes the runtime APIs are invoked directly as express middlewares.
+    // In order to make sure the "this" keyword always have the correct context,
+    // we bind every single method to have the context of the object itself
+    for (const property in runtime) {
+        if (runtime[property] instanceof Function) {
+            runtime[property] = runtime[property].bind(runtime)
+        }
+    }
```

Comment:
⚠️ `for...in` iterates all properties of an object, including inherited properties like `toString` and `valueOf`. We don't need to do that, and probably don't want to. (Not sure it's _bad_ bad, just should generally be avoided.)

Reply:
Good call, let me double check. If this is iterating over unnecessary properties then we should change it!

Reply:
ok, I've added a console.log in the for...in loop and this is the console output:

```
_configure
_logStartupMessage
_getProtocol
_getDefaultCacheControl
_strictSSL
_setCompression
_setupLogging
_setupMetricsFlushing
_updatePackageMobify
_configureProxyConfigs
_createApp
_createExpressApp
_addSDKInternalHandlers
_setupSSRRequestProcessorMiddleware
_setupProxying
_setupHealthcheck
_setupCommonMiddleware
_validateConfiguration
_addStaticAssetServing
_addDevServerGarbageCollection
serveServiceWorker
serveStaticFile
render
_createHandler
createHandler
_getRequestProcessor
_redirectToLoadingScreen
_getDevServerHostAndPort
_getDevServerURL
_getWebpackAsset
```

It doesn't look like it loops over built-in Object methods like `toString` or `valueOf`. So i think we should be good?